File: heart/src/main/java/com/vantagecircle/heartrate/component/AppComponent.java
package com.vantagecircle.heartrate.component;

import android.app.Application;
import android.content.Context;

import com.vantagecircle.heartrate.HeartApplication;
import com.vantagecircle.heartrate.data.DataManager;
import com.vantagecircle.heartrate.data.DatabaseHelper;
import com.vantagecircle.heartrate.annotation.ApplicationContext;
import com.vantagecircle.heartrate.module.AppModule;

import javax.inject.Singleton;

import dagger.Component;

/**
 * Created by bapidas on 09/10/17.
 */
@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    void inject(HeartApplication heartApplication);
    @ApplicationContext
    Context getContext();
    Application getApplication();
    DataManager getDataManager();
    DatabaseHelper getDatabaseHelper();
}

File: heart/src/main/java/com/bapidas/heartrate/di/component/AppComponent.kt
package com.bapidas.heartrate.di.component

import android.app.Application
import android.content.Context
import com.bapidas.heartrate.HeartApplication
import com.bapidas.heartrate.data.DataManager
import com.bapidas.heartrate.data.DatabaseHelper
import com.bapidas.heartrate.di.annotation.ApplicationContext
import com.bapidas.heartrate.di.module.AppModule
import dagger.Component
import javax.inject.Singleton

/**
 * Created by bapidas on 09/10/17.
 */
@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {
    fun inject(heartApplication: HeartApplication?)

    @get:ApplicationContext
    val context: Context
    val application: Application
    val dataManager: DataManager
    val databaseHelper: DatabaseHelper
}